community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -458,10 +460,20 @@ private void commit() throws TransactionFailureException
 
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
-            // Trigger transaction "before" hooks
-            if ( (hooksState = hooks.beforeCommit( txState, this, storeLayer )) != null && hooksState.failed() )
+            // Trigger transaction "before" hooks.
+            if ( hasChanges() )
```

Comment:
`TransactionEventHandler` describes methods that can be used to listen certain type of _transaction_ events, and for example transaction can be committed irrespectively to operation types that it just performed. 
I also can see that in java doc it says that read only transactions will not be notified and previous behaviour was a bug according to that description, but still that sounds strange to me.
I still can imagine use cases when people would like to be notified when transaction was committed or rollbacked disregard disregard it was read only.